linera-views/src/indexed_db.rs

```diff
@@ -0,0 +1,359 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{collections::BTreeMap, fmt::Debug};
+
+use indexed_db_futures::{js_sys, prelude::*, web_sys};
+use thiserror::Error;
+
+use crate::{
+    batch::{Batch, WriteOperation},
+    common::{
+        get_upper_bound_option, CommonStoreConfig, ContextFromStore, LocalAdminKeyValueStore,
+        LocalKeyValueStore, LocalReadableKeyValueStore, LocalWritableKeyValueStore,
+    },
+    test_utils::generate_test_namespace,
+    value_splitting::DatabaseConsistencyError,
+    views::ViewError,
+};
+
+/// The initial configuration of the system
+#[derive(Debug)]
+pub struct IndexedDbStoreConfig {
+    /// The common configuration of the key value store
+    pub common_config: CommonStoreConfig,
+}
+
+impl IndexedDbStoreConfig {
+    /// Creates a `IndexedDbStoreConfig`. `max_concurrent_queries` and `cache_size` are not used.
+    pub fn new(max_stream_queries: usize) -> Self {
+        let common_config = CommonStoreConfig {
+            max_concurrent_queries: None,
+            max_stream_queries,
+            cache_size: 1000,
+        };
+        Self { common_config }
+    }
+}
+
+/// The number of streams for the test
+pub const TEST_INDEX_DB_MAX_STREAM_QUERIES: usize = 10;
+
+const DATABASE_NAME: &str = "linera-views";
```

Comment:
That seems inadequate. Within the framework of this library, all databases could claim to be `linera-views`. Why not use `IndexedDB`?

Reply:
Within IndexedDB, all databases can claim to be IndexedDB :)  This key is namespaced within IndexedDB, not within `linera-views` — though actually, since IndexedDB is automatically isolated between domains, this whole string is essentially arbitrary.